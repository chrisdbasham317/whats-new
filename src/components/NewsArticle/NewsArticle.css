.section--news-cards {
  align-items: center;
  background-color: #ffffff;
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  height: 500px;
  max-width: 48%;
  margin-bottom: 20px;
}

.img--card-pic {
  height: auto;
  max-height: 50%;
  min-width: 100%;
  object-fit: stretch;
}