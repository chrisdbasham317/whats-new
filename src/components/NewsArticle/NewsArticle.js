import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({ id, headline, img, description, url }) => {
  return <section className='section section--news-cards'>
    <img alt='' className='img img--card-pic' data={id} src={img} />
    <h2>{headline}</h2>
    <article>{description}</article>
    <footer data={url}>Link to Article</footer>
  </section>
}

export default NewsArticle;