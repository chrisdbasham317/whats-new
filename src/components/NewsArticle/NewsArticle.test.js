import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  let wrapper
  
  beforeEach(() => {
    wrapper = shallow(<NewsArticle
      key='3'
      headline='Bananas'
      img='this is an image'
      description='this is a description'
      url= 'this is a url'
      />);
    });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})