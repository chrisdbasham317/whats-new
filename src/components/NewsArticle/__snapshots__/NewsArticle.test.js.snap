// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot 1`] = `
<section
  className="section section--news-cards"
>
  <img
    alt=""
    className="img img--card-pic"
    src="this is an image"
  />
  <h2>
    Bananas
  </h2>
  <article>
    this is a description
  </article>
  <footer
    data="this is a url"
  >
    <a
      href="this is a url"
    >
      Link to Article
    </a>
  </footer>
</section>
`;
